modules/wallet/seed.go

```diff
@@ -216,3 +241,79 @@ func (w *Wallet) LoadSeed(masterKey crypto.TwofishKey, seed modules.Seed) error
 	defer w.mu.Unlock()
 	return w.loadSeed(masterKey, seed)
 }
+
+// SweepSeed scans the blockchain for outputs generated from seed and create a
+// transaction that transfers them to the wallet. Note that this incurs a
+// transaction fee. It returns the total value of the outputs, minus the fee.
+// TODO: support SiafundOutputs too
+func (w *Wallet) SweepSeed(masterKey crypto.TwofishKey, seed modules.Seed) (payout types.Currency, err error) {
+	if err = w.tg.Add(); err != nil {
+		return
+	}
+	defer w.tg.Done()
+
+	if !w.cs.Synced() {
+		return types.Currency{}, errors.New("cannot sweep until blockchain is synced")
+	}
+
+	// get an address to spend into
+	var uc types.UnlockConditions
+	err = w.db.Update(func(tx *bolt.Tx) error {
+		var err error
+		uc, err = w.nextPrimarySeedAddress(tx)
+		return err
+	})
+	if err != nil {
+		return
+	}
+
+	// scan blockchain for outputs, filtering out 'dust' (outputs that cost
+	// more in fees than they are worth)
+	s := newSeedScanner(seed)
+	_, maxFee := w.tpool.FeeEstimation()
+	s.dustThreshold = maxFee.Mul64(350) // fee is per byte; output+signature is approx. 350 bytes
+	if err = s.scan(w.cs); err != nil {
+		return
+	}
+
+	// construct a transaction that spends the outputs
+	// TODO: this may result in transactions that are too large.
+	var txn types.Transaction
+	var swept types.Currency // total value of swept outputs
+	for _, output := range s.siacoinOutputs {
+		// construct an input for the output
+		sk := generateSpendableKey(seed, output.seedIndex)
+		txn.SiacoinInputs = append(txn.SiacoinInputs, types.SiacoinInput{
+			ParentID:         types.SiacoinOutputID(output.id),
+			UnlockConditions: sk.UnlockConditions, // TODO: check timelock, etc.
+		})
+		// add a signature for the input
+		swept = swept.Add(output.value)
+	}
+
+	estTxnSize := len(s.siacoinOutputs) * 350 // since we use maxFee, lowballing is ok
+	estFee := maxFee.Mul64(uint64(estTxnSize))
```

Comment:
ah, I see what you mean. I used "est" to indicate that it was derived from the estimated size of the transaction. It could be renamed to "sweepFee" or "txnFee" but I don't think it matters much.